.idx-templates/latest/idx-template.nix

```diff
@@ -1,21 +1,57 @@
-{
-  pkgs,
-  astroTemplate ? "basics",
-  ...
-}:
-{
-  packages = [
-    pkgs.nodejs_20
-  ];
+{ pkgs, astroTemplate ? "basics", debug ? false, ... }: {
+  packages = [ pkgs.git pkgs.nodejs_20 pkgs.nodePackages.pnpm ];
```

Comment:
Nix should respect `COREPACK` that way we not only provide the binaries but also determine this using the `packageManger` in the root `package.json` to specify versions along with `engine` would be more than sufficient to allow pnpm to take over as the defacto package manager for the env
```
env = {
    COREPACK_ENABLED = "1";
  };
  ```
  
  This is to allow nix to respect `COREPACK` should you wish to try

Reply:
The corepack entrypoint and shims are on a separate Nix derivation, which we can enable on the `dev.nix` for sure. This file is for creating the project, I don't think corepack helps with anything here.

Reply:
good point, would you be open to enabling it in `dev.nix` ?

Reply:
Happy to consider it but let’s keep this PR focused on making things work and we can definitely iterate in future PRs!

Reply:
it works a treat 💚